build-logic/jvm/src/main/kotlin/gradlebuild.unittest-and-compile.gradle.kts

```
@@ -318,3 +319,10 @@ fun TaskContainer.registerCITestDistributionLifecycleTasks() {
         group = ciGroup
     }
 }
+
+// https://github.com/gradle/gradle-private/issues/3380
+fun Test.configureAndroidUserHome() {
+    val androidUserHomeForTest = project.layout.buildDirectory.asFile.get().resolve("androidUserHomeForTest/$name").absolutePath
```

```suggestion
    val androidUserHomeForTest = project.layout.buildDirectory.dir("androidUserHomeForTest/$name").get().asFile.absolutePath
```

This could be an alternative. Or can we use `Provider`s for environment variables? Then you could use `project.layout.buildDirectory.dir("androidUserHomeForTest/$name")` directly.

---

I guess not...